test/jdk/java/math/BigInteger/MutableBigIntegerShiftTests.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.test.lib.RandomFactory;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.Arguments;
+
+import java.math.BigInteger;
+import java.math.MutableBigIntegerBox;
+import java.util.Arrays;
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static java.math.MutableBigIntegerBox.*;
+
+/**
+ * @test
+ * @bug 8336274
+ * @summary Tests for correctness of MutableBigInteger.leftShift(int)
+ * @library /test/lib
+ * @build jdk.test.lib.RandomFactory
+ * @build java.base/java.math.MutableBigIntegerBox
+ * @key randomness
+ * @run junit MutableBigIntegerShiftTests
+ */
+public class MutableBigIntegerShiftTests {
+
+    private static final int ORDER_SMALL = 60;
+    private static final int ORDER_MEDIUM = 100;
+
+    private static final Random random = RandomFactory.getRandom();
+
+    private static int[] orders() {
+        return new int[] { ORDER_SMALL, ORDER_MEDIUM };
+    }
+
+    @ParameterizedTest
+    @MethodSource("orders")
+    public void shift(int order) {
+        for (int i = 0; i < 100; i++) {
+            test(fetchNumber(order), random.nextInt(200));
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource("pathTargetedCases")
+    public void test(MutableBigIntegerBox x, int n) {
+        leftShiftAssertions(x, n);
```

Comment:
One last thing... It would make sense to inline `leftShiftAssertions()` and remove it?

Reply:
I preferred to pur the assertions in a different method, in case that tests for `MBI.rightShift()` would be added in future.